### PR TITLE
Updating checkToken usage to utilise the new JControllerLegacy::checkToken

### PR DESCRIPTION
**Pull request by izharaazmi**
Pull Request in continuation to #8445 

### Summary of Changes
Implementing the new pattern everywhere. All controllers that extend `JControllerLegacy` are set to use the "nice" handling instead of white screen on a token validation failure. 

Wherever this was within a JSON response, the handling is kept intact as they are better suited it those contexts, changing only the message text.

### Testing Instructions
Test the various backend and front-end forms which would earlier show a blank white page with just a message "Invalid Token". They should now redirect to same page if possible otherwise to home page; with appropriate message.

To see the effect the respective forms should be submitted after manipulating the token input in the form so as to trigger invalid token situation. This can usually be done by selecting in the browser "inspect element" (or similar) and modify the DOM.

If the token is valid (not manipulated or expired), everything should be working as usual without any difference.

The affected components are:
* Banners
* Cache
* Categories
* Contacts
* ContentHistory
* CustomFields
* Finder
* Installer
* Joomla Update
* Language Manager
* Login/Logout
* Media Manager
* Module Manager
* Menu Manager
* Tags
* Template Manager
* User Manager
* Joomla Installation Application (`/installation`)

 
### Documentation Changes Required
Developer documentation should include this feature introduction. Text can be taken from #8445

### CODE REVIEW

File: administrator/components/com_installer/controllers/update.php [hunk 1]
```diff
@@ -81,7 +81,7 @@ public function update()
 	 */
 	public function find()
 	{
-		(JSession::checkToken() or JSession::checkToken('get')) or jexit(JText::_('JINVALID_TOKEN'));
+		$this->checkToken('request');
```
Comment (andrepereiradasilva) [inline administrator/components/com_installer/controllers/update.php:79]: Why the change to request?

Comment (izharaazmi) [reply administrator/components/com_installer/controllers/update.php:79]: `JSession::checkToken() or JSession::checkToken('get')` checks for `post` first and then fallbacks to `get`. Hence I simplified it to a single call that would match from either super-global.
Is that not appropriate?

Comment (andrepereiradasilva) [reply administrator/components/com_installer/controllers/update.php:79]: EDITED: yeah seems fine. i guess

File: components/com_config/controller/cmsbase.php [hunk 2]
```diff
@@ -33,8 +33,12 @@ class ConfigControllerCmsbase extends JControllerBase
...
 	public function execute()
 	{
-		// Check for request forgeries
-		JSession::checkToken() or jexit(JText::_('JINVALID_TOKEN'));
+		// Check for request forgeries.
+		if (!JSession::checkToken())
+		{
+			$this->app->enqueueMessage(JText::_('JINVALID_TOKEN_NOTICE'));
+			$this->app->redirect('index.php');
+		}
```
Comment (andrepereiradasilva) [inline components/com_config/controller/cmsbase.php:41]: why is this one different?

File: administrator/components/com_installer/controllers/update.php (hunk 1 above)
Comment (izharaazmi) [reply administrator/components/com_installer/controllers/update.php:79]: Currently it uses `jexit` we're trying to avoid it here.
My approach was not translating the statement. rather I tried to follow the motive of doing `post` then `get` check. I think it was done so that both can be accepted. `request` does the same... hence.
Maybe someone else can give better insight then.

**Comment by andrepereiradasilva** [comment]
ok from code review, just added one question, when possible will try to do some tests and see if there is no other left.

File: components/com_config/controller/cmsbase.php (hunk 2 above)
Comment (izharaazmi) [reply components/com_config/controller/cmsbase.php:41]: This doesn't inherit from `JControllerLegacy` so can't use `$this->checkToken` here. Skipped referer check to avoid too much inline code.

**Comment by izharaazmi** [comment]
There is something wrong in all merge 'staging' happening these days. Not sure why. Will look again into this later.

**Comment by SamuelSchepp** [comment]
I have tested this item :red_circle: unsuccessfully on 64c78f18671127a25a9157b0981ded8dc07f3c63<br /><br />Unable to apply patch via git.
Applying patch with com_patchtester did not resolve the issues.

Tested:
- Banners (New -> Name "test" -> removed last character of token -> Save)
- Categories (New -> Name "test" -> removed last character of token -> Save)
- Module Manager (New -> „Articles - Latest“ -> Title „test“ -> removed last character of token -> Save)

Result:
- Banners: White Error Page
- Categories: White Error Page
- Module Manager: New message (green box)

Result: No change in Banners (admin) und Categories (admin).
@icampus

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12801">issues.joomla.org/tracker/joomla-cms/12801</a>.</sub>

**Comment by izharaazmi** [comment]
@SamuelSchepp There are conflicts so patchtester/git may not be able to apply patch correctly. I'll resolve the conflicts then please test again.

**Comment by sabaturgay** [comment]
I have tested this item :white_check_mark: successfully on 578a42f089f8601134fcd925b52a0b646d11635c<br /><br />I have tested this item succesfully 
I changed session lifetime of global configuration to 1 minute

Tested:
Banners (New -> Name "test" -> wait 1 minute -> Save)
Contacts(New -> Name "test" -> wait 1 minute -> Save)


Result:
Banners: redirected sign in page
Contacts: redirected sign in page
@icampus

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12801">issues.joomla.org/tracker/joomla-cms/12801</a>.</sub>

**Comment by bene-we** [comment]
I have tested this item :white_check_mark: successfully on 578a42f089f8601134fcd925b52a0b646d11635c<br /><br />What I tested:

Banner -> new -> Name "Test" -> Clear Site Data -> Save
Contacts -> new -> Name "Test" -> Clear Site Data -> Save
Joomla! Update -> Choose File -> Clear Site Data -> Upload & Install


Results:

Banner: Redirect to login page, after login redirect to Banner List page
Contacts: Redirect to login page, after login redirect to Contacts List page
Joomla! Update: Redirect to login page, after login redirect to home page<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12801">issues.joomla.org/tracker/joomla-cms/12801</a>.</sub>

@icampus

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

File: administrator/components/com_installer/controllers/languages.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+defined('_JEXEC') or die;
+
+use Joomla\Utilities\ArrayHelper;
+
+/**
+ * Languages Installer Controller
+ *
+ * @since  2.5.7
+ */
+class InstallerControllerLanguages extends JControllerLegacy
```
Comment (mbabker) [inline administrator/components/com_installer/controllers/languages.php]: Umm, is this new file intended to be part of this PR?

**Comment by roland-d** [comment]
@mbabker Good catch. This is not a new file but a file that was deleted in commit 204ef46ae742a8ab7c2604c8d02e020f49d65837 and this PR has not been updated since. So that is why it is still there. So this file should be removed from this PR. Either by @izharaazmi or a maintainer merging this PR.

**Comment by ghost** [comment]
@izharaazmi can you please remove the File mentioned above by @roland-d and @mbabker?

**Comment by izharaazmi** [comment]
Done the conflict anomaly fix. I have updated 5 more of those usages. So maintainers might recommend to clear current test status. 

PS: Can we have a new Label something like "Needs Work" or "Author Attention" so that it can be found out clearly which PR needs attention?

**Comment by ghost** [comment]
> Label something like "Needs Work" or "Author Attention"

@izharaazmi at Issue Tracker is an "Needs Review"-Status. But this isn't shown here and vice versa (Issue about this is open https://github.com/joomla/jissues/issues/1032).

**Comment by izharaazmi** [comment]
IMO, "Needs Review" sound more like a **Reviewer attention** rather than **Author attention**. Or not?

**Comment by ghost** [comment]
True. It was an Information about what is available.
